lib/in_out/reader.py

```diff
@@ -7,48 +7,48 @@
 import collections
 from collections import defaultdict
 
-def linux_input(log_directory, channel_list, starting_date, ending_date):
+def linux_input(log_directory, channels_requested, starting_date, ending_date):
     """  
         reads the IRC-Logs files in linux environment
 
     Args:
         log_directory (str): Location of the logs (Assumed to be arranged in directory structure as : <year>/<month>/<day>/<log-file-for-channel>.txt)
         here log_directory will have the directory where all the folders 2013 2014 2015 etc are stored
-        channels_list (list): list of channels to be perform analysis on OR one element list ["ALL"] to perform analyis on all channels in the folder
+        channels_requested (list): list of channels to be perform analysis on OR one element list ["ALL"] to perform analyis on all channels in the folder
         starting_date(str): Starting date in the format yyyy-mm-dd
         ending_date(str)  : Ending date in the format yyyy-mm-dd
 
     Returns:
         logs (dict) : dictionary with key(str) as 'yyyy-mm-dd' and value as another dictionary {"data":day_data,"channel":channel_name}
 
     """
-    def check_if_path_and_file_is_valid(work_path, channel_list):
+    def check_if_path_and_file_is_valid(work_path, channels_requested):
         if not os.path.exists(work_path):
             raise IOError("Path " + work_path + " doesn't exist")
```

Comment:
The test needs to check for IOError case.

Reply:
@tsrkp Sir, I have already added a test for the IOError case in PR #270. Please see [this](https://github.com/prasadtalasila/IRCLogParser/blob/dev/test/unit-test/test_lib/test_in_out/test_reader.py#L66).